library/src/main/kotlin/com/chuckerteam/chucker/api/ChuckerCollector.kt

```diff
@@ -74,23 +81,50 @@ public class ChuckerCollector @JvmOverloads constructor(
      * @param context Application context
      * @param startTimestamp The timestamp to read transactions from. Passing null means
      * transactions will not be limited by timestamp
+     * @param exportFormat The export format: LOG or HAR
      * @return The content uri of a file with the transactions in or null if the export failed.
      */
+    @SuppressLint("VisibleForTests")
     public fun writeTransactions(
         context: Context,
         startTimestamp: Long?,
+        exportFormat: ExportFormat = ExportFormat.LOG
     ): Uri? {
         val transactions =
             RepositoryProvider.transaction().getTransactionsInTimeRange(startTimestamp)
         if (transactions.isEmpty()) {
             return null
         }
 
-        val sharableTransactions = TransactionListDetailsSharable(transactions, encodeUrls = false)
-        return sharableTransactions.writeToFile(
-            context = context,
-            fileName = "api_transactions.txt",
-        )
+        return when (exportFormat) {
+            ExportFormat.LOG -> {
+                val sharableTransactions =
+                    TransactionListDetailsSharable(transactions, encodeUrls = false)
+
+                sharableTransactions.writeToFile(
+                    context = context,
+                    fileName = "api_transactions.txt",
+                )
```

Comment:
Let's refactor this block out as it's repeated

Reply:
it actually changes the extension but ill refactor